lib/cartopy/mpl/patch.py

```diff
@@ -208,6 +212,8 @@ def path_to_geos(path, force_ccw=False):
                 isinstance(geom, Polygon) and
                 collection[-1][0].contains(geom.exterior)):
             collection[-1][1].append(geom.exterior)
+        elif isinstance(geom, Point):
+            other_result_geoms.append(geom)
```

Comment:
Yep. It's not perfect, as we lose some of the ordering, but it fits the requirements for now.